### PR TITLE
EAPI 8 eclass updates

### PR DESCRIPTION
**Pull request by dm0-**
@mgorny Here are the last few eclass updates.  The `meson-multilib.eclass` change has a dependency on #21409.

Aside from just adding EAPI 8 to the list of supported EAPIs, most of these changes fall into:
  - Dropping support for EAPIs that are not supported in inherited eclasses.
  - Moving the EAPI check before the inherit guard so using newer bash functionality won't result in syntax errors for old versions.
  - Setting the inherit guard variable before inherits so it prevents loops.
  - Trivial formatting unification where different styles are mixed in the same file.

@mattst88 @floppym @SoapGentoo

### CODE REVIEW

File: eclass/xdg-utils.eclass
Comment (floppym) [inline eclass/xdg-utils.eclass:99]:
```suggestion
		elif [[ -e ${dir}/icon-theme.cache ]]; then
```

File: eclass/systemd.eclass
Comment (floppym) [inline eclass/systemd.eclass:235]: We should ban the systemd tmpfiles functions in EAPI 8.

**Review by floppym — CHANGES_REQUESTED** [review]
Left comments for systemd and xdg-utils.

meson-multilib and udev look ok.

File: eclass/xdg-utils.eclass
Comment (dm0-) [reply eclass/xdg-utils.eclass:99]: Is that the intent of that condition?  I thought it was testing if `icon-theme.cache` is the *only* file in the directory.

File: eclass/systemd.eclass
Comment (dm0-) [reply eclass/systemd.eclass:235]: Should they be banned, or just removed?  I asked about [bug #740638](https://bugs.gentoo.org/740638) in the gentoo-devel e-mail: is there a removal date?

File: eclass/xdg-utils.eclass
Comment (floppym) [reply eclass/xdg-utils.eclass:99]: Ah, I see. Never mind then.

File: eclass/systemd.eclass
Comment (floppym) [reply eclass/systemd.eclass:235]: Yeah, let's remove them in that case.

Comment (dm0-) [reply eclass/systemd.eclass]: Rebased with that removal in its own commit.

**Comment by dm0-** [comment]
I'll drop the deprecated tmpfiles removal here after #21462 is merged.

File: eclass/autotools.eclass
Comment (ulm) [inline eclass/autotools.eclass:4]: I was told by its maintainers that there are some other changes for autotools.eclass pending, which should be done together with the EAPI 8 update. So I'd suggest that you drop it from the PR for now.

File: eclass/xdg-utils.eclass
Comment (ulm) [inline eclass/xdg-utils.eclass:10]: Drop EAPIs 0 to 4? Keeping them doesn't make much sense, especially since `xdg.eclass` no longer supports 0 to 3 either.
```suggestion
# @SUPPORTED_EAPIS: 5 6 7 8
```

File: eclass/xdg-utils.eclass
Comment (ulm) [inline eclass/xdg-utils.eclass:21]:
```suggestion
case ${EAPI} in
	5|6|7|8) ;;
```

File: eclass/xdg-utils.eclass
Comment (ulm) [inline eclass/xdg-utils.eclass:96]: For consistency, either do
```suggestion
				fails+=("${dir}")
```
here, or keep the blank in `local fails=( )` above.

**Review by ulm — CHANGES_REQUESTED** [review]

**Review by ulm — APPROVED** [review]
LGTM

File: eclass/autotools.eclass
Comment (ulm) [reply eclass/autotools.eclass]: I know. Time frame is about one month (or so I was told).

**Comment by dm0-** [comment]
I dropped the deprecation cleanup commit since #21462 was merged and the conflict was resolved.

**Review by mgorny — APPROVED** [review]
LGTM.

**Pull request closed without merging**
